apps/sms/js/drafts.js

```diff
@@ -32,183 +26,107 @@
     get size() {
       return draftIndex.size;
     },
+
     /**
-     * add
-     *
-     * Push a Draft object or an object that has
-     * all the properties of a Draft instance
-     * to the Drafts collection. If the object
-     * isn't an instance of Draft, initialize
-     * a new Draft.
-     *
-     * @param  {Object}  draft  Draft-like object.
-     *
-     * @return {Drafts} return the Drafts object.
+     * Pushes a Draft instance to the Drafts in-memory collection and commits
+     * it to the persistent storage.
+     * @param  {Draft} draft Draft instance.
+     * @returns {Drafts} Returns the Drafts object.
      */
     add: function(draft) {
-      var threadId;
-      var thread;
-      var stored;
-
-      if (draft) {
-        if (!(draft instanceof Draft)) {
-          draft = new Draft(draft);
-        }
+      if (!draft || !(draft instanceof Draft)) {
+        throw new Error('Draft should be valid Draft instance.');
+      }
 
-        threadId = draft.threadId || null;
-        thread = draftIndex.get(threadId) || [];
-        stored = thread[thread.length - 1];
+      // Remove existing draft.
+      this.delete(draft);
```

Comment:
```delete``` also checks type of the draft, I added check above just for consistency, not sure if it makes code more readable...